_includes/assets/js/search.js

```diff
@@ -1,115 +1,118 @@
-var indicatorSearch = function(inputElement, indicatorDataStore) {
-  that = this;
-  this.inputElement = inputElement;
-  this.indicatorDataStore = indicatorDataStore;
-  this.indicatorData = [];
-  this.hasErrored = false;
-
-  this.processData = function(data) {
-    for(var goalLoop = 0; goalLoop < data.length; goalLoop++) {
-      for(var indicatorLoop = 0; indicatorLoop < data[goalLoop].goal.indicators.length; indicatorLoop++) {
-        var currentIndicator = data[goalLoop].goal.indicators[indicatorLoop];
-        currentIndicator.goalId = data[goalLoop].goal.id;
-        currentIndicator.goalTitle = data[goalLoop].goal.title;
-        that.indicatorData.push(currentIndicator);
-      }
+var indicatorSearch = function() {
+  // Helper function to make a search query "fuzzier", using the ~ syntax.
+  var getFuzzierQuery = function(query, fuzziness) {
+    return query
+      .split(' ')
+      .map(function(x) { return x + '~' + fuzziness; })
+      .join(' ');
+  }
+  // Helper function to get the matched words from a result set.
+  var getMatchedTerms = function(results) {
+    var matchedTerms = {};
+    results.forEach(function(result) {
+      Object.keys(result.matchData.metadata).forEach(function(matchedTerm) {
+        matchedTerms[matchedTerm] = true;
+      })
+    });
+    return Object.keys(matchedTerms);
+  }
+  // Helper function to get a boost score, if any.
+  var getSearchFieldOptions = function(field) {
+    var opts = {}
+    if (opensdg.searchIndexBoost[field]) {
+      opts['boost'] = intval(opensdg.searchIndexBoost[field])
     }
-  };
+    return opts
+  }
+  var urlParams = new URLSearchParams(window.location.search);
+  var searchTerms = urlParams.get('q');
+  if (searchTerms) {
```

Comment:
Be careful with using equality like this one in Javascript. `if(searchTerms)` will return false if the query is `undefined` (as you'd expect here) but it will also return false if the search term is a string containing `"0"`. It'd be better if you were more explicit with your expected search term. (Search for "javascript falsy" for more info)